### PR TITLE
New output columns from TBProfiler Task

### PR DESCRIPTION
**Pull request by cimendes**
## Motivation

This PR adds a new behaviour, controlled by the `tbprofiler_additional_outputs` boolean (currently set to `true`) that controls 5 new outputs: `tb_profiler_variant_gene_name`, `tb_profiler_variant_locus_tag`, `tb_profiler_variant_substitutions`,  `tb_profiler_sequencing_method`, `tbprofiler_additional_outputs_csv` and `tbprofiler_laboratorian_csv`. 

These columns include information in all mutations found by tbprofiler with the following criteria:

- who catalogue associated with resistance, uncertain, and interim (not associated with resistance not included)
- mutation is not synonymous
- ALL mutations in **katG, pncA, rpoB, ethA and gid** included, except for synonymous mutations  

## Main changes

- tbprofiler container image was updated to `staphb/tbprofiler:4.4.2` 
- The following new output columns were included with information about each mutation in a comma-delimited list to be ingested into a BigQuery database for visualization in a Looker Dashboard:
  - **Column1:** tbprofiler_variant_gene_name
  - **Column2:** tbprofiler_variant_locus_tag
  - **Column3:** tbprofiler_variant_substitutions in the format `mutation_type:nt_sub(aa_sub` 
  - **Column4:** tbprofiler_sequencing_method
  - **Column5:** CSV file to be ingested into CDPH LIMS system with all the information above split into different columns
  - **Column6:** CSV file to be human readable with the same information as above, and additionally the depth of coverage, the frequency, and a warning if the coverage is below a threshold of 10 times. 

## Testing

- `tbprofiler_additional_outputs` set to `true`
  - [Theiaprok_Illumina_SE with 100 samples](https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/c29d2ac7-177a-48a8-90c9-30f681cd4d03)
  - [Theiaprok_Illumina_PE with 100 samples](https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/7a99d3c6-4136-4f3e-8bf1-8b2fa91cff8b)

### CODE REVIEW

File: tasks/species_typing/task_tbprofiler.wdl [hunk 1]
```diff
@@ -30,7 +33,7 @@ task tbprofiler {
       INPUT_READS="-1 ~{read1} -2 ~{read2}"
     fi
 
-    # Run Kleborate on the input assembly with the --all flag and output with samplename prefix
+    # Run tb-profiler on the input reads with samplename prefix
```
Comment (frankambrosio3) [inline tasks/species_typing/task_tbprofiler.wdl:36]: thanks  you!

**Comment by cimendes** [comment]
Test with commit hash [1af02af](https://github.com/theiagen/public_health_bacterial_genomics/pull/217/commits/1af02af456024353b90ce74162d7efa18c8e0cef)

- 100 PE Samples: https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/0392d1d4-8999-453d-b07f-df79fa777de8
- 100 SE Samples: https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/2c06b0ae-7be9-4b46-8c13-f05e37aa837b

File: tasks/species_typing/task_tbprofiler.wdl [hunk 2]
```diff
@@ -75,6 +88,92 @@ task tbprofiler {
...
+    
+        # laboratorian report
+        with open("tbprofiler_laboratorian_report.csv", "wt") as report_fh:
+          report_fh.write("tbprofiler_gene_name,tbprofiler_locus_tag,tbprofiler_variant_substitutions,confidence,depth,frequency,warning\n")
+          for i in range(0, len(gene_name)):
+            if not depth[i]:  # for cases when depth is null, it gets converted to 0
+              depth[i] = 0
+            warning = "Low depth coverage" if  depth[i] < int('~{min_depth}') else "" # warning when coverage is lower than the defined 'min_depth' times
+            report_fh.write(gene_name[i] + ',' + locus_tag[i] + ',' + variant_substitutions[i] + ',' + confidence[i] + ',' + str(depth[i]) + ',' + str(frequency[i]) + ',' + warning + '\n')
+
```
Comment (frankambrosio3) [inline tasks/species_typing/task_tbprofiler.wdl:176]: This looks great. Noting this section for future reuse. As we add lims integration we'll want to generate files like these again. Nice work!!!

File: tasks/species_typing/task_tbprofiler.wdl [hunk 3]
```diff
@@ -89,14 +188,20 @@ task tbprofiler {
...
+    String tbprofiler_gene_name = read_string("GENE_NAME")
+    String tbprofiler_locus_tag = read_string("LOCUS_TAG")
+    String tbprofiler_variant_substitutions = read_string("VARIANT_SUBSTITUTIONS")
+    String tbprofiler_output_seq_method_type = read_string("OUTPUT_SEQ_METHOD_TYPE")
   }
   runtime {
     docker: "~{tbprofiler_docker_image}"
     memory: "16 GB"
-    cpu: 8
+    cpu: cpu
```
Comment (frankambrosio3) [inline tasks/species_typing/task_tbprofiler.wdl:201]: Nice

**Pull request merged**